Rewrite README

wjt opened the pull request:
Apparently I never updated this to talk about ISO images. There were
many other missing details that I added while I was here.

dbnicholson commented on README.md:
```diff
@@ -1,45 +1,130 @@
 
-Modes
------
+The expected partition layout in this case is: EFI and BIOS boot partitions, an
```
Is this still accurate after #87? As far as I can tell, it's just an EFI partition + a root partition of some type.

wjt replied:
Actually this is still technically correct because it refers to the partition layout of the `eosinstaller` image, not the OS image it is installing, but I'll reword to make this clearer.

dbnicholson reviewed (commented):
Thanks, this is very helpful. I just had a question about a particular.

wjt committed:
Rewrite README

Apparently I never updated this to talk about ISO images. There were
many other missing details that I added while I was here.
